sdk/test-utils/perf/test/event/mockEventHubConsumerClient.ts

```diff
@@ -89,14 +97,25 @@ export class MockEventReceiver2 {
     processEvents: (events: Event[], context: { partitionId: number }) => Promise<void>,
     partitionId: number
   ) {
+    const startTime = process.hrtime();
+    let eventsRaised = 0;
     while (this.closeCalled === false) {
-      let numberOfEvents = this.getRandomInteger(1, this.maxBatchSize);
-      const events: Event[] = [];
-      while (numberOfEvents--) events.push({ body: generateUuid() });
-      await this.processFuncWithDelay(
-        async () => processEvents(events, { partitionId }),
-        this.getRandomInteger(this.minDelay, this.maxDelay)
-      );
+      const elapsed = process.hrtime(startTime);
+      const elapsedSeconds = elapsed[0] + elapsed[1] / 1000000000;
+      const targetEventsRaised = elapsedSeconds * this.maxEventsPerSecondPerPartition;
+
+      if (eventsRaised < targetEventsRaised) {
```

Comment:
This class needs two codepaths.  A "throttled mode" when `maxEventsPerSecond` is set, and an "unthrottled mode" when `maxEventsPerSecond` is not set (or set to a sentinel value like `0` or `-1`).  This will allow to us to measure two things:

1. If the client can raise events infinitely fast, how quickly can the perf framework process them?
2. If the client raises events at exactly `maxEventsPerSecond`, does the perf framework measure the same number.

Reply:
updated